src/NodeCompiler/CompilerContext.php

```diff
@@ -41,16 +54,40 @@ public function getSelf() : ?ReflectionClass
             throw new \RuntimeException('The current context does not have a class for self');
         }
 
-        return $this->self;
+        return $this->context;
     }
 
     public function getReflector() : Reflector
     {
         return $this->reflector;
     }
 
+    /**
+     * @return ReflectionClass|ReflectionFunction
+     * @throws \RuntimeException
+     */
+    public function getContext()
```

Comment:
This is the same as already defined `getSelf()` apart from signature. I see it's done to avoid BC breaks, but it makes me uneasy to duplicate up like this :/ not sure best approach here (this may already be the best approach)